cap_build.c

```diff
@@ -0,0 +1,68 @@
+#include "cheriintrin.h"
+#include "include/common.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+int main()
+{
+    // We create two capabilities: x, and y, which is derived from x. We note that, though we use
+    // `cheri_address_set()`, it doesn't actually set (i.e. _change_) the address of its arguments,
+    // but construct a new capability with the address we pass.
+    //
+    // cheri_address_set(void* __capability, uintptr_t) -> void *__capability
+    void *__capability x = cheri_ddc_get();
+    uintptr_t y_ptr = (uintptr_t) malloc(20);
+    void *__capability y = cheri_address_set(x, y_ptr);
+
+    printf("\n\tThe second capability printed below is created from the first using");
+    printf(
+        "\n\t`cheri_address_set()`. Notice that `cheri_address_set()` created a new capability,");
+    printf("\n\trather than setting the address on the first.");
+    printf("\n\n");
```

Comment:
I would change these prints into comments, so on executing, we just have the capability prints, and then can refer to side-by-side with the example code. It's more that this is how all examples (to my knowledge) are done.

Reply:
Sorry for the delay. Fixed in [be6d139](https://github.com/capablevms/cheri-examples/pull/92/commits/be6d139256ad63fa4067d244d8082f8926efe9fd) — thanks!